### PR TITLE
Display name for applications

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:
Follow up for https://github.com/kubermatic/kubermatic/pull/13331

**Which issue(s) this PR fixes**:

xref https://github.com/kubermatic/kubermatic/issues/13330

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
`DisplayName` for applications is used on the UI
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: modules/web/src/app/shared/components/application-list/component.ts [hunk 1]
```diff
@@ -188,6 +188,10 @@ export class ApplicationListComponent implements OnInit, OnDestroy {
     return this.applicationDefinitionsMap.get(name)?.spec?.logoData;
   }
 
+  getApplicationName(name: string): SafeUrl {
```
Comment (Waseem826) [inline modules/web/src/app/shared/components/application-list/component.ts:191]:
```suggestion
  getApplicationName(name: string): string {
```

File: modules/web/src/app/shared/components/application-list/template.html [hunk 2]
```diff
@@ -104,10 +104,10 @@
               <img *ngIf="getApplicationLogo(application.spec.applicationRef.name) as logo; else applicationDefName"
                    class="km-application-logo"
                    [src]="logo"
-                   [alt]="application.spec.applicationRef.name"
+                   [alt]="getApplicationName(application.spec.applicationRef.name)"
```
Comment (Waseem826) [inline modules/web/src/app/shared/components/application-list/template.html:107]: Should we also use this in `matTooltip` below?

**Review by Waseem826 — COMMENTED** [review]
Couple of small nit picks. PTAL

File: modules/web/src/app/shared/components/application-list/template.html (hunk 2 above)
Comment (ahmedwaleedmalik) [reply modules/web/src/app/shared/components/application-list/template.html:107]: I was thinking of leaving the tool tip to point to the actual name but can change it as well. WDYT?

Comment (Waseem826) [reply modules/web/src/app/shared/components/application-list/template.html:107]: I think its fine if we leave it to `name`.

File: modules/web/src/app/shared/components/application-list/component.ts (hunk 1 above)
Comment (ahmedwaleedmalik) [reply modules/web/src/app/shared/components/application-list/component.ts]: Updated

**Review by Waseem826 — APPROVED** [review]
/approve

**Review by Waseem826 — APPROVED** [review]

**Review by Waseem826 — APPROVED** [review]

**Pull request merged**